### DISCUSSION TITLE
How to do scratch training or finetuning for Speaker diarization on my own dataset

### DISCUSSION THREAD

**Discussion by saumyaborwankar** [NVIDIA-NeMo/NeMo#4086] [Q&A] [answered]
Please help, I was able to find documentation regarding making my own dataset in nemo format but couldnt find any material to use that dataset to train a model or finetune a pretrained model.

**Comment by nithinraok** [comment]
Are you planning to train a speaker model or VAD model or both for speaker diarization?

**Comment by saumyaborwankar** [comment]
Ideally i want to train both.

**Comment by nithinraok** [comment, marked as answer]
**For speaker embedding:**
Prepare your manifest using these steps:
https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/asr/speaker_recognition/datasets.html#all-other-datasets 

Then Prepare the hydra configuration file as detailed here:
https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/asr/speaker_recognition/configs.html#dataset-configuration 

You may refer to [this section](https://github.com/NVIDIA/NeMo/tree/main/examples/speaker_tasks/recognition#training) as well on how to use the training script with hydra configuration file. 

**For VAD:**
Prepare your manifest using these steps: 
https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/asr/speech_classification/datasets.html#speech-command-freesound-for-vad 

Then Prepare the hydra configuration file as detailed here:
https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/stable/asr/speech_classification/configs.html 

Then use this [script](https://github.com/NVIDIA/NeMo/blob/main/examples/asr/speech_classification/speech_to_label.py) to train a VAD model. 


Inference for speaker diarization:

For speaker diarization inference using trained models refer to [these steps](https://github.com/NVIDIA/NeMo/tree/main/examples/speaker_tasks/diarization#run-speaker-diarization-on-your-audio-files).

**Reply by manchandasahil** [reply]
Hi @nithinraok , Thanks for those instructions. I am looking to start training speaker embedding model on phone conversations. For this training, Should each wav file contain an audio from one speaker? and should the length of each audio be cut to less than 5 seconds? Do we have to do this pre processing before we start ?
